modules/core/src/effects/lighting/lighting-effect.js

```diff
@@ -62,14 +62,13 @@ export default class LightingEffect extends Effect {
     if (this.directionalLights.some(light => light.shadow)) {
       this.shadow = true;
       this._addShadowModule();
+      this._createLightMatrix();
```

Comment:
Light props (direction, position, etc.) may change after the effect is created. Can you move this back and add a comment so that future maintainers remember.